### ISSUE TITLE
Give template ability to modify KssStyleguide before generate()

### ISSUE THREAD

**Issue by jonmilner** [kss-node/kss-node#245]
Is it currently possible to use markdown in combination with a custom comment property, e.g. `--custom foo`?

What I'd like to do is something like this:

```
/*
Buttons

Buttons use the class `.button`, which can be applied to an `<a>`, `<button>`,
or `<input>`.

Guidelines:
There are three standard types of buttons:
* Floating action button: A circular material button that lifts and displays an ink reaction on press.
* Raised button: A typically rectangular material button that lifts and displays ink reactions on press.
* Flat button: A button made of ink that displays ink reactions on press but does not lift.

Markup:
<a class="button" role="button">Link</a>
<button class="button" type="submit">Button</button>
<input class="button" type="button" value="Input">
<input class="button" type="submit" value="Submit">

Styleguide Components.Buttons
*/
```

```
{{#if description}}
  <div class="kss-description">{{{description}}}</div>
{{/if}}
{{#if guidelines}}
  <div class="kss-guidelines">{{{guidelines}}}</div>
{{/if}}
```
I'm using `--custom guidelines` to style content blocks that provide usability guidelines, rather than the description, which I reserve for development related instructions / notes.

The problem is that markdown doesn't appear to work in the custom comment block. In the example provided above, the content will display like this...

> There are three standard types of buttons: * Floating action button: A circular material button that lifts and displays an ink reaction on press. * Raised button: A typically rectangular material button that lifts and displays ink reactions on press. * Flat button: A button made of ink that displays ink reactions on press but does not lift.

... when I'd like it to display like this...

> There are three standard types of buttons:
* Floating action button: A circular material button that lifts and displays an ink reaction on press.
* Raised button: A typically rectangular material button that lifts and displays ink reactions on press.
* Flat button: A button made of ink that displays ink reactions on press but does not lift.

Can this be done?

**Comment by JohnAlbin** [comment]
Right now we tell the parser about custom properties. While we could tell the parser to run a callback on the contents of the custom property (so you could render the markdown), I don't think that would be useful in the general case of "I want to do something with the custom property."

The only way to ensure the most flexibility is to give the template a chance to modify the parsed KssStyleguide object before the generator makes the HTML pages.

**Comment by JohnAlbin** [comment]
Hmm… now that I've thought this through, this functionality can be fixed with #182.
